adoc/extensions/sycl_khr_command_graph.adoc

```diff
@@ -252,111 +253,423 @@ a@ Constructs a [code]#depends_on_all_leaves# property instance.
+
+.[apititle]#khr::command_graph::begin_recording#
+[source, role=synopsis,id=api:khr-command-graph-begin-recording]
 ----
-namespace sycl::khr::property::graph {
-class assume_buffer_outlives_graph {
-  public:
-    assume_buffer_outlives_graph() = default;
-};
-}
+void begin_recording(queue& recordingQueue,                     (1)
+                       const property_list& propList = {})
+
+void begin_recording(const std::vector<queue>& recordingQueues, (2)
+                       const property_list& propList = {})
+
 ----
+_Constraints:_ [code]#State# is [code]#graph_state::modifiable#.
+
+_Effects (1):_ Synchronously changes the state of [code]#recordingQueue# to the
+[code]#queue_state::recording# state.
 
-==== Enable-Profiling Property [[enable-profiling]]
+_Effects (2):_ Synchronously changes the state of each queue in
+[code]#recordingQueues# to the [code]#queue_state::recording# state.
 
-[source,c++]
+This function accepts a [code]#property_list# parameter for future use but
+currently there are no properties defined for it in this extension.
+
+_Throws (1)(2):_
+
+* An [code]#exception# with the [code]#errc::invalid# error code if any queue is
+  currently recording to a graph.
+
+* An [code]#exception# with the [code]#errc::invalid# error code if the device
+  or context associated with any queue is not the same as those used to create
+  the graph.
+
+
+.[apititle]#khr::command_graph::end_recording#
+[source, role=synopsis,id=api:khr-command-graph-end-recording]
 ----
-namespace sycl::khr::graph {
-class enable_profiling {
-  public:
-    enable_profiling() = default;
-};
-}
+void end_recording()                                          (1)
+
+void end_recording(queue& recordingQueue)                     (2)
+
+void end_recording(const std::vector<queue>& recordingQueues) (3)
+----
+_Constraints:_ [code]#State# is [code]#graph_state::modifiable#.
+
+_Effects (1):_ Synchronously finishes recording on all queues that are recording
+to the graph and sets their state to [code]#queue_state::executing#.
+This operation is a no-op for any queue in the graph that is already in the
+[code]#queue_state::executing# state.
+
+_Effects (2):_ Synchronously changes the state of [code]#recordingQueue# to the
+[code]#queue_state::executing# state.
+This operation is a no-op if [code]#recordingQueue# is already in the
+[code]#queue_state::executing# state.
+
+_Effects (3):_ Synchronously changes the state of each queue in
+[code]#recordingQueues# to the [code]#queue_state::executing# state.
+This operation is a no-op for any queue in [code]#recordingQueues# that is
+already in the [code]#queue_state::executing# state.
+
+_Throws (2)(3):_
+
+* An [code]#exception# with the [code]#errc::invalid# error code if any queue is
+recording to a different graph.
+
+.[apititle]#khr::command_graph::add (empty node)#
+[source, role=synopsis,id=api:khr-command-graph-add-empty]
+----
+node add(const property_list& propList = {})
```

Comment:
For the enqueue commands KHR (https://github.com/KhronosGroup/SYCL-Docs/pull/644) it's being proposed internally that we move to a handler-less interface, effectively removing the need for command groups. If we do go this route and this is adopted in the KHR it may be suggested that for SYCL graphs we also move away from using handlers.

I don't think we need to do anything immediately for this but something to be aware of.

Reply:
Good to know! Yeah I think we can evaluate that when the design is finalized. I think John (?) said he was going to present it to the WG soon (this week maybe?).